nilearn/datasets/atlas.py

```diff
@@ -369,14 +380,22 @@ def fetch_atlas_harvard_oxford(atlas_name, data_dir=None,
         resume=resume,
         verbose=verbose)
     atlas_niimg = check_niimg(atlas_img)
+    label_idx_to_names = {k: v for k, v in enumerate(names)}
     if not symmetric_split or is_lateralized:
+        if not is_probabilistic and clean_labels:
+            valid_idx = np.unique(get_data(atlas_niimg))
+            names = [label_idx_to_names[i] for i in valid_idx]
```

Comment:
if we do this we break the association between image values and labels that the region where image == i corresponds to labels[i], no?

Reply:
@jeromedockes Yes this is true. The documentation doesn't say clearly that integers from the image are supposed to be interpreted as indices in the list of labels though. Unless I am missing something, this is not the case for all atlases, right? Some have values that are not even consecutive as @bthirion mentioned earlier.

Reply:
I also agree that using dictionaries for such mappings would be much easier.